### PR TITLE
[MM-61458] Fix errcheck issues in server/channels/app/plugin_requests.go

### PR DESCRIPTION
**Pull request by molejnik88**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/29073
Jira https://mattermost.atlassian.net/browse/MM-61458

#### Release Note

```release-note

```

### CODE REVIEW

File: server/channels/app/plugin_requests.go [hunk 1]
```diff
@@ -147,17 +151,19 @@ func (ch *Channels) servePluginRequest(w http.ResponseWriter, r *http.Request, h
...
-		if session != nil && err == nil && cookieAuth && r.Method != "GET" {
+		if session != nil && appErr == nil && cookieAuth && r.Method != "GET" {
 			sentToken := ""
 
 			if r.Header.Get(model.HeaderCsrfToken) == "" {
 				bodyBytes, _ := io.ReadAll(r.Body)
 				r.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
-				r.ParseForm()
+				if err := r.ParseForm(); err != nil {
+					mlog.Warn("Failed to parse form data", mlog.Err(err))
```
Comment (hanzei) [inline server/channels/app/plugin_requests.go:165]:
```suggestion
					mlog.Warn("Failed to parse form data for plugin request", mlog.Err(err))
```

**Review by hanzei — COMMENTED** [review]
One small request, otherwise LGTM

File: server/channels/app/plugin_requests.go (hunk 1 above)
Comment (molejnik88) [reply server/channels/app/plugin_requests.go]: Surę, will do so.

**Review by marianunez — APPROVED** [review]

**Review by hanzei — APPROVED** [review]
Thank you!

**Pull request merged**